lib/state.js

```diff
@@ -0,0 +1,66 @@
+const debug = require('debug')('snyk:nodejs:state');
+
+const state = {
+  events: [],
+  filters: {},
+  packages: {},
+};
+
+module.exports = {
+  get,
+  clean,
+  addEvent,
+  addFilter,
+  addPackage,
+  removeFilter,
+};
+
+function get() {
+  return state;
+}
+
+function clean() {
```

Comment:
Why not cleanEvents?

Reply:
right now it only cleans the events, that's right, but this function's purpose is to "clean the state"
so a more explicit name might be "cleanState" but since we're in the `state` module already, it's just "clean".
what do you think?

Reply:
Maybe `transmissionState`? Something that indicates that this is the place to aggregate things to be sent

Reply:
well, it doesn't necessarily have to be things to be sent.

I have half a mind to move some of the things the debugger tracks there. which right now might be a bad idea.
and I have some ideas of moving the read snapshot in there as well. which might not be amazing either.

one of the reasons I'm removing these things from the transmitter into state is to decouple what the agent knows about itself and what it decides to send out